packages/aws-cdk-lib/aws-synthetics/lib/canary.ts

```diff
@@ -513,11 +543,45 @@ export class Canary extends cdk.Resource implements ec2.IConnectable {
   }
 
   private createRunConfig(props: CanaryProps): CfnCanary.RunConfigProperty | undefined {
-    if (!props.environmentVariables) {
+    if (props.activeTracing === undefined &&
+      !props.environmentVariables &&
+      !props.memory &&
+      !props.timeout) {
       return undefined;
     }
+    const activeTracingNotSupportedRuntime = [
+      Runtime.SYNTHETICS_PYTHON_SELENIUM_2_1,
+      Runtime.SYNTHETICS_PYTHON_SELENIUM_3_0,
+    ];
+
+    if (props.activeTracing && activeTracingNotSupportedRuntime.includes(props.runtime)) {
+      throw new Error('You can enable active tracing only for canaries that use version syn-nodejs-2.0 or later for their canary runtime.');
+    }
+
+    let memoryInMb = undefined;
+    if (!cdk.Token.isUnresolved(props.memory) && props.memory) {
+      memoryInMb = props.memory.toMebibytes();
+      if (memoryInMb % 64 !== 0) {
+        throw new Error(`\`memory\` must be a multiple of 64 MiB, got ${memoryInMb} MiB.`);
+      }
+      if (memoryInMb < 960 || memoryInMb > 3008) {
+        throw new Error(`\`memory\` must be between 960 MiB and 3008 MiB, got ${memoryInMb} MiB.`);
+      }
+    }
+
+    let timeoutInSeconds = undefined;
+    if (!cdk.Token.isUnresolved(props.timeout) && props.timeout) {
```

Comment:
According to the [documentation](https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-runconfig.html#cfn-synthetics-canary-runconfig-timeoutinseconds), the property can't set this time to be longer than the frequency of the runs of this canary.

However, there are various ways to set the `schedule` property, and I determined that implementing this validation would be difficult. Therefore, I decided not to implement it. If you have any better suggestions or methods, please let me know.